### ISSUE TITLE
LHN - Employee name is displayed as workspace chat name

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#54661]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.79-2
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** N/A
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team
**Component:** Left Hand Navigation (LHN)

## Action Performed:
Preconditions:
The user uses two accounts, 1 Workspace Admin and 1 employee
1.Go to staging.new.expensify.com
2.Log in with the employee account and with admin account on another device
3. As an admin, invite an employee to the Workspace
4. As an employee, observe the LHN

## Expected Result:
After being invited to a workspace, the WS name is displayed in the LHN for the employee

## Actual Result:
The employee name is displayed as workspace chat name on LHN, only after navigating to the chat, the correct WS name appears

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details> 

https://github.com/user-attachments/assets/2a071408-614b-4ec5-bcf1-36a89ed92fd1

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by FitseTLT** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
LHN - Employee name is displayed as workspace chat name

### What is the root cause of that problem?
When the policy expense chat arrives `isOwnPolicyExpenseChat` is becoming false until the user opens the report so this condition fails https://github.com/Expensify/App/blob/12bbe02d606f8aa01359b9603a2308872d3bb3e1/src/libs/ReportUtils.ts#L2965-L2966
### What changes do you think we should make in order to solve the problem?

For these cases, we can give a fallback of checking if ownerAccountID is the current user account id
```
 if (report?.isOwnPolicyExpenseChat || report?.ownerAccountID === currentUserAccountID) {
        return getPolicyName(report, false, policy);
```
### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?

We can create a test for getPolicyExpenseChatName to check that it returns the proper title for own policy chat though isOwnPolicyExpenseChat is not populated correctly
### What alternative solutions did you explore? (Optional)

**Comment by huult** [comment]
I think this issue is related to https://github.com/Expensify/App/issues/54327 because `isOwnPolicyExpenseChat` is `false`, and `chatType` is `policyExpenseChat`.

**Comment by RachCHopkins** [comment]
I can definitely replicate this. @chiragsalian I see you deemed https://github.com/Expensify/App/issues/54327 as BE - is this the same? And does this need to be a separate GH issue, or will it be fixed organically with that one?

**Comment by chiragsalian** [comment]
Yeah looks like the fix for https://github.com/Expensify/App/issues/54327 should resolve this one. And its an internal backend issue. Closing this issue in favor of this one https://github.com/Expensify/App/issues/54327.
